CourseProject/src/main/java/com/neoflex/CourseProject/service/ValidationService.java

```diff
@@ -0,0 +1,103 @@
+package com.neoflex.CourseProject.service;
+
+import com.neoflex.CourseProject.dto.LoanStatementRequestDto;
+
+import com.neoflex.CourseProject.exception.ValidationException;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Service
+public class ValidationService {
+
+    public Boolean resultsOfValidation(LoanStatementRequestDto loanStatementRequestDto) throws ValidationException {
+        boolean resultsOfValidations=true;
+       Boolean[] testResults;
+       if(loanStatementRequestDto.getMiddleName() != null){
+           int maxCountConditions = 9;
+           testResults = new Boolean[maxCountConditions];
+           testResults[8] = isNamesValid(loanStatementRequestDto.getMiddleName());
+       }else{
+           int minCountConditions = 8;
+           testResults = new Boolean[minCountConditions];
+       }
+       testResults[0] = isEmailValid(loanStatementRequestDto.getEmail());
+       testResults[1] = isNamesValid(loanStatementRequestDto.getFirstName());
+       testResults[2] = isNamesValid(loanStatementRequestDto.getLastName());
+       testResults[3] = isPassportSeriesValid(loanStatementRequestDto.getPassportSeries());
+       testResults[4] = isPassportNumberValid(loanStatementRequestDto.getPassportNumber());
+       testResults[5] = isAmountMeetsTheCondition(loanStatementRequestDto.getAmount());
+       testResults[6] = isTermMeetsTheCondition(loanStatementRequestDto.getTerm());
+       testResults[7] = isAdult(loanStatementRequestDto.getBirthdate());
+       return resultsOfValidations;
+   }
+
+    Boolean isEmailValid(String email) throws ValidationException {
+       String mailRegex = "^[A-Za-z0-9+_.-]+@(.+)$";
+       Pattern pattern = Pattern.compile(mailRegex);
+       Matcher matcher = pattern.matcher(email);
+       if(matcher.matches())
```

Comment:
Чтобы избавиться от дублирования постоянного Pattern и Matcher можешь воспользоваться встроенным методом у String:
```suggestion
       if (email.matches(mailRegex)) {
           return true;
       } else {
           throw new ValidationException("email is not valid " + email);
       }
```